crates/router/src/core/refunds_v2.rs

```diff
@@ -0,0 +1,798 @@
+
+#[allow(clippy::too_many_arguments)]
+#[instrument(skip_all)]
+pub async fn trigger_refund_to_gateway(
+    state: &SessionState,
+    refund: &storage::Refund,
+    merchant_account: &domain::MerchantAccount,
+    key_store: &domain::MerchantKeyStore,
+    payment_attempt: &storage::PaymentAttempt,
+    payment_intent: &storage::PaymentIntent,
+) -> RouterResult<storage::Refund> {
+    let db = &*state.store;
+
+    let mca_id = payment_attempt
+        .merchant_connector_id
+        .clone()
+        .ok_or(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to retrieve mca id from payment attempt")?;
+
+    let storage_scheme = merchant_account.storage_scheme;
+    metrics::REFUND_COUNT.add(
+        1,
+        router_env::metric_attributes!(("connector", mca_id.get_string_repr().to_string())),
+    );
+
+    let mca = db
+        .find_merchant_connector_account_by_id(&state.into(), &mca_id, key_store)
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to fetch merchant connector account")?;
+
+    let connector_enum = mca.connector_name;
+
+    let connector: api::ConnectorData = api::ConnectorData::get_connector_by_name(
+        &state.conf.connectors,
+        &connector_enum.to_string(),
+        api::GetToken::Connector,
+        Some(mca_id.clone()),
+    )?;
+
+    refunds_validator::validate_for_valid_refunds(payment_attempt, connector.connector_name)?;
+
+    let mut router_data = core_utils::construct_refund_router_data(
+        state,
+        connector_enum,
+        merchant_account,
+        key_store,
+        payment_intent,
+        payment_attempt,
+        refund,
+        &mca,
+    )
+    .await?;
+
+    let add_access_token_result =
+        access_token::add_access_token(state, &connector, merchant_account, &router_data, None)
+            .await?;
+
+    logger::debug!(refund_router_data=?router_data);
+
+    access_token::update_router_data_with_access_token_result(
+        &add_access_token_result,
+        &mut router_data,
+        &payments::CallConnectorAction::Trigger,
+    );
+
+    let router_data_res = if !(add_access_token_result.connector_supports_access_token
+        && router_data.access_token.is_none())
+    {
+        let connector_integration: services::BoxedRefundConnectorIntegrationInterface<
+            api::Execute,
+            types::RefundsData,
+            types::RefundsResponseData,
+        > = connector.connector.get_connector_integration();
+        let router_data_res = services::execute_connector_processing_step(
+            state,
+            connector_integration,
+            &router_data,
+            payments::CallConnectorAction::Trigger,
+            None,
+        )
+        .await;
+        let option_refund_error_update =
+            router_data_res
+                .as_ref()
+                .err()
+                .and_then(|error| match error.current_context() {
```

Comment:
#Do From transformation instead directly writing the hard coding the data in parent functions, below also